### PR TITLE
Allow features to enable store auth use.

### PR DESCRIPTION
**Pull request by saukko**
[ssu] Allow features to enable store auth use. Fixes JB#32346

Previously store-auth-repos could be only defined in repos.ini, with
this patch features can define themselves the need for use store-auth.
This removes the need to touch repos.ini when bringing in new feature.

Signed-off-by: Marko Saukko <marko.saukko@jolla.com>

### CODE REVIEW

**Comment by saukko** [comment]
Changed so that now feature defines domains that store auth should be used for that feature.

File: libssu/ssu.cpp [hunk 1]
```diff
@@ -107,6 +108,18 @@ QString Ssu::credentialsScope(QString repoName, bool rndRepo){
   if (storeAuthRepos.contains(repoName))
     return "store";
 
+  // If we defined explicitly what credentials to use with which secure domain
+  // use those.
+  QHash<QString, QString> secureDomainAuth;
+  SsuVariables::variableSection(&repoSettings,"secure-domain-auth", &secureDomainAuth);
```
Comment (rainemak) [inline libssu/ssu.cpp]: Nit: missing space between ```&repoSettings,"secure-domain-auth"```

File: libssu/ssu.cpp [hunk 2]
```diff
@@ -107,6 +108,18 @@ QString Ssu::credentialsScope(QString repoName, bool rndRepo){
...
+  QHashIterator<QString, QString> i(secureDomainAuth);
+  while (i.hasNext()) {
+    i.next();
+    if (repoUrl(repoName,rndRepo).contains(i.key()) && !i.value().isEmpty()) {
```
Comment (rainemak) [inline libssu/ssu.cpp]: Nit: missing space between ```repoName,rndRepo```

**Comment by kananoja** [comment]
LGTM, but I'm not very familiar with the SSU code. Maybe @bwachter could check this?

**Comment by mkosola** [comment]
LGTM
---
[INFO] 16:01:46 Test set: ssu
[INFO] 16:01:46 Starting test case ut_ssu
[INFO] 16:01:46 Finished test case ut_ssu Result: PASS
---

**Pull request merged**